src/contracts/middleware/IndexRegistry.sol

```diff
@@ -199,26 +206,54 @@ contract IndexRegistry is IIndexRegistry {
 
         // if the operator is not the last in the list, we must swap the last operator into their positon
-        if(operatorId != operatorIdToSwap){
+        if (operatorId != operatorIdToSwap) {
             //update the swapped operator's operatorIdToIndexHistory list with a new entry, as their index has now changed
             _updateOperatorIdToIndexHistory(operatorIdToSwap, quorumNumber, indexOfOperatorToRemove);
         } 
         // marking the final entry in the deregistering operator's operatorIdToIndexHistory entry with the deregistration block number
         _operatorIdToIndexHistory[operatorId][quorumNumber][_operatorIdToIndexHistory[operatorId][quorumNumber].length - 1].toBlockNumber = uint32(block.number);
     }
 
-    /**
-     * @notice remove an operator from the globalOperatorList  
-     * @param indexOfOperatorToRemove index of the operator to remove
-     */ 
-    function _removeOperatorFromGlobalOperatorList(uint32 indexOfOperatorToRemove) internal {
-        uint32 globalOperatorListLastIndex = uint32(globalOperatorList.length - 1);
-        bytes32 operatorIdToSwap;
-        if(indexOfOperatorToRemove != globalOperatorListLastIndex){
-            operatorIdToSwap = globalOperatorList[globalOperatorListLastIndex];
-            globalOperatorList[indexOfOperatorToRemove] = operatorIdToSwap;
-            emit GlobalIndexUpdate(operatorIdToSwap, indexOfOperatorToRemove);
+    /// @notice Returns the total number of operators of the service for the given `quorumNumber` at the given `blockNumber`
+    function _getTotalOperatorsForQuorumAtBlockNumber(uint8 quorumNumber, uint32 blockNumber) internal view returns (uint32){
+        // if there are no entries in the total operator history, return 0
+        if (_totalOperatorsHistory[quorumNumber].length == 0) {
+            return 0;
         }
-        globalOperatorList.pop();
+        
+        // if there is only one entry in the total operator history, return it
+        if (_totalOperatorsHistory[quorumNumber].length == 1) {
+            return _totalOperatorsHistory[quorumNumber][_totalOperatorsHistory[quorumNumber].length - 1].index;
+        }
+
+        // loop backwards through the total operator history to find the total number of operators at the given block number
+        uint256 totalOperatorsHistoryLength = _totalOperatorsHistory[quorumNumber].length;
+        for (uint i = 0; i <= totalOperatorsHistoryLength - 2; i++) {
+            uint256 index = totalOperatorsHistoryLength - 2 - i;
+            OperatorIndexUpdate memory totalOperatorUpdate = _totalOperatorsHistory[quorumNumber][index];
+            if (totalOperatorUpdate.toBlockNumber <= blockNumber) {
+                return _totalOperatorsHistory[quorumNumber][index + 1].index;
+            }
+        }        
+        return _totalOperatorsHistory[quorumNumber][0].index;
```

Comment:
OK, am going to suggest a rewrite:
```
    /// @notice Returns the total number of operators of the service for the given `quorumNumber` at the given `blockNumber`
    /// @dev Returns zero if the @param blockNumber is from before the @param quorumNumber existed, and returns the current number of total operators if the @param blockNumber is in the future.
    function _getTotalOperatorsForQuorumAtBlockNumber(uint8 quorumNumber, uint32 blockNumber) internal view returns (uint32){
// store list length in memory
uint256 totalOperatorsHistoryLength = _totalOperatorsHistory[quorumNumber].length;
        // if there are no entries in the total operator history, return 0
        if (totalOperatorsHistoryLength == 0) {
            return 0;
        }
        
// if `blockNumber` is in future, return current number of operators
if (blockNumber > uint32(block.number)) {
return _totalOperatorsHistory[quorumNumber][totalOperatorsHistoryLength - 1].index;
// else if `blockNumber` is from before the `quorumNumber` existed, return `0`
} else if (`blockNumber` < _totalOperatorsHistory[quorumNumber][0].fromBlockNumber {
return 0;
}

        // loop backwards through the total operator history to find the total number of operators at the given block number
        for (uint256 i = 0; i <= totalOperatorsHistoryLength - 1; i++) {
            uint256 listIndex = (totalOperatorsHistoryLength - 1) - i;
            OperatorIndexUpdate memory totalOperatorUpdate = _totalOperatorsHistory[quorumNumber][listIndex];
            // look for the first update that began at or after `blockNumber`
            if (totalOperatorUpdate.fromBlockNumber >= blockNumber) {
                return _totalOperatorsHistory[quorumNumber][listIndex].index;
            }
        }        
        return _totalOperatorsHistory[quorumNumber][0].index;
```

I believe this has all the properties it should.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/81/commits/57d8aca9b8c542f7781985eb7dd78b38e9069956